packages/nova-react-test-utils/README.md

```diff
@@ -2,4 +2,170 @@
 
 The Nova Facade is a set of interfaces that represent the core framework dependencies of a data-backed UI component in a large application. This allows high value components to be written in a host agnostic fashion and used within any host that implements the Nova contracts.
 
-This package provides test utilities for components written with the React specific implementation.
+This package provides test utilities for components written with the React specific implementation. The purpose of these utilities is to expose capabilities of [MockPayloadGenerator](https://github.com/microsoft/graphitation/tree/main/packages/graphql-js-operation-payload-generator) and [Apollo mock client](https://github.com/microsoft/graphitation/tree/main/packages/apollo-mock-client) in Nova context.
+
+## Unit tests
+
+To leverage the power of mentioned packages inside your unit tests, you need to wrap the tested component with `NovaMockEnvironmentProvider`:
+
+```tsx
+import {
+  createMockEnvironment,
+  MockPayloadGenerator,
+  NovaMockEnvironmentProvider,
+} from "@nova/react-test-utils";
+import { render, screen, act } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+
+const environment = createMockEnvironment(schema);
+render(
+  <NovaMockEnvironmentProvider environment={environment}>
+    <SomeTestSubject />
+  </NovaMockEnvironmentProvider>,
+);
+
+await act(async () =>
+  environment.graphql.mock.resolveMostRecentOperation((operation) =>
+    MockPayloadGenerator.generate(operation),
+  ),
+);
+
+const button = await screen.findByRole("button", { name: "Click me" });
+userEvent.click(button);
+expect(environment.eventing.bubble).toHaveBeenCalled();
+```
+
+Let's break it down:
+
+`createMockEnvironment` creates a Nova environment object that can be used with the `NovaMockEnvironmentProvider` and has mocks instantiated for each piece of the facade interface (eventing, commanding and graphql). `createMockEnvironment` takes a schema as an argument. The schema is a GraphQL schema that is later used to generate mock data for the GraphQL mock.
+
+The `environment.graphql.mock` mock provides an API to act on operations issued by a component tree. For details on the API see [apollo-mock-client docs](https://github.com/microsoft/graphitation/tree/main/packages/apollo-mock-client). Together with mock client, which intercepts all operations, for mock data generation one can use `MockPayloadGenerator` whose generate function is leveraging [graphql-js-operation-payload-generator](https://github.com/microsoft/graphitation/tree/main/packages/graphql-js-operation-payload-generator).
```

Comment:
The docs for these need to be created/updated, but in the meantime it _might_ be helpful to link to https://relay.dev/docs/guides/testing-relay-components/

I'm on the fence, though. What do you think?

Reply:
Hmm fair point, to at least mention it's built on top of it. I just didn't like relay docs because they rely heavily on `enzyme`.

Reply:
Updated, let me know what you think :)

Reply:
Yup, looks great 👌